infra/staging/main.tf

```diff
@@ -0,0 +1,137 @@
+###############
+## Variables ##
+###############
+
+variable "project_id" {
+  type = string
+  default = "call-home-staging"
+}
+
+variable "location" {
+  type = string
+  default = "asia-southeast1"
+}
+
+variable "app_bucket_name" {
+  type = string
+  default = "call-home-staging-strapi-app-bucket"
+}
+
+variable "db_name" {
+  type = string
+  default = "strapi"
+}
+
+variable "db_instance_name" {
+  type = string
+  default = "strapi-db-instance"
+}
+
+variable "db_username" {
+  type = string
+  sensitive = true
+}
+
+variable "db_password" {
+  type = string
+  sensitive = true
+}
+
+######################
+## Project Settings ##
+######################
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
+
+// Enable Cloud SQL Admin API to connect to database
+resource "google_project_service" "sql_admin_api" {
+  project = data.google_project.project.project_id
+  service = "sqladmin.googleapis.com"
+}
+
+// Enable Cloud Build API to allow building for App Engine
+resource "google_project_service" "cloud_build_api" {
+  project = data.google_project.project.project_id
+  service = "cloudbuild.googleapis.com"
+}
+
+// Enable App Engine permissions for Cloud Build service account
+resource "google_project_iam_binding" "app_engine_admin" {
+  project = data.google_project.project.project_id
+  role = "roles/appengine.appAdmin"
+  members = [
+    "serviceAccount:${data.google_project.project.number}@cloudbuild.gserviceaccount.com"
+  ]
+  depends_on = [
+    google_project_service.cloud_build_api
+  ]
+}
+
+// Enable Service Account user permissions for Cloud Build service account
+resource "google_project_iam_binding" "service_account_user" {
+  project = data.google_project.project.project_id
+  role = "roles/iam.serviceAccountUser"
+  members = [
+    "serviceAccount:${data.google_project.project.number}@cloudbuild.gserviceaccount.com"
+  ]
+  depends_on = [
+    google_project_service.cloud_build_api
+  ]
+}
+
+#############
+## Storage ##
+#############
+
+resource "google_storage_bucket" "app_bucket" {
+  name = var.app_bucket_name
+  location = var.location
+  project = data.google_project.project.project_id
+  force_destroy = true
+  versioning {
+    enabled = true
+  }
+}
+
+################
+## App Engine ##
+################
+
+resource "google_app_engine_application" "app" {
+  project = data.google_project.project.project_id
+  location_id = var.location
+}
+
+##############
+## Database ##
+##############
+
+resource "google_sql_database_instance" "strapi_db_instance" {
+  name = var.db_instance_name
+  database_version = "POSTGRES_14"
+  project = data.google_project.project.project_id
+  region = var.location
+  deletion_protection = false
+  settings {
+    tier = "db-f1-micro"
+    maintenance_window {
+      day = 7
+      hour = 20 # Sunday 8pm (UTC) = Monday 4am (UTC+8)
+    }
+  }
+}
+
+resource "google_sql_database" "strapi_db" {
```

Comment:
Wow, Terraform names. TIL `google_sql_database_instance` is the thing that costs compute resources and `google_sql_database` is an SQL database INSIDE the database_instance. How clear /s.

I assume we'll configure this database name in Strapi?

Reply:
yep, the database name will be configured for the Strapi app through env vars in [app.template.yaml](https://github.com/glencbz/call-home-cms-experiments/pull/6/files#diff-a63b4136b40f69b3e3523db48febd1c844cbbfe61bc68fd0e6d82fc80f7c31c8)!